File: LeetCode/C/226-tree-翻转二叉树.cpp
/*
 * @lc app=leetcode.cn id=226 lang=cpp
 *
 * [226] 翻转二叉树
 */
#include "main.h"
// @lc code=start
class Solution {
  public:
    TreeNode *invertTree(TreeNode *root) {
        invert(root);
        return root;
    }

    void invert(TreeNode *root) {
        if (root == nullptr)
            return;
        swap(root->left, root->right);
        invert(root->left);
        invert(root->right);
    }
};
// @lc code=end

File: LeetCode/Go/226-tree-翻转二叉树.go
/*
 * @lc app=leetcode.cn id=226 lang=golang
 *
 * [226] 翻转二叉树
 */
package leetcode

// @lc code=start
func invertTree(root *TreeNode) *TreeNode {
	if root == nil {
		return nil
	}
	root.Left = invertTree(root.Left)
	root.Right = invertTree(root.Right)
	root.Left, root.Right = root.Right, root.Left
	return root
}

// @lc code=end
